src/Form/QueueContentsForm.php

```diff
@@ -0,0 +1,173 @@
+<?php
+
+namespace Drupal\media_mpx\Form;
+
+use Drupal\Core\Batch\BatchBuilder;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\media_mpx\Repository\MpxMediaType;
+use Drupal\media_mpx\Service\QueueVideoImportsRequest;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Class QueueContentsForm.
+ */
+class QueueContentsForm extends FormBase {
+
+  /**
+   * The mpx Media Types repository.
+   *
+   * @var \Drupal\media_mpx\Repository\MpxMediaType
+   */
+  private $mpxMediaTypeRepository;
+
+  /**
+   * QueueContentsForm constructor.
+   */
+  public function __construct(MpxMediaType $mpxMediaTypeRepository) {
+    $this->mpxMediaTypeRepository = $mpxMediaTypeRepository;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    $mpx_media_types = $this->mpxMediaTypeRepository->findAllTypes();
+    $options = [];
+    foreach ($mpx_media_types as $mpx_media_type) {
+      $options[$mpx_media_type->id()] = $mpx_media_type->label();
+    }
+
+    $form['mpx_video_types'] = [
+      '#type' => 'checkboxes',
+      '#title' => t('mpx Video Types'),
+      '#options' => $options,
+      '#required' => TRUE,
+    ];
+    $form['queue'] = [
+      '#type' => 'submit',
+      '#value' => $this->t('Queue selected Video Types'),
```

Comment:
We're not queuing the _types_ specifically, but the videos associated with each type. How about just `Import videos`?

Reply:
Yeah, clearly English my is brightest not today at its. Changed!